internal/compiler-interface/src/main/java/xsbti/compile/ExternalHooks.java

```diff
@@ -51,6 +51,8 @@
          * @return API changes
          */
         boolean shouldDoIncrementalCompilation(Set<String> changedClasses, CompileAnalysis previousAnalysis);
+
+        Optional<FileHash[]> hashClasspath(File[] classpath);
```

Comment:
Kind of curious. Does it matter if this takes `File[]` or just `hash(File)`?

Reply:
This way you can add e.g. paralelizm but with `hash(File)` you are not.

Reply:
ok. Sounds good to me.